src/account/ModuleManagerInternals.sol

```diff
@@ -227,6 +229,16 @@ abstract contract ModuleManagerInternals is IModuleManager {
         }
     }
 
+    function _onInstall(address module, bytes calldata data, bytes4 interfaceId) internal {
+        if (data.length > 0) {
+            if (!ERC165Checker.supportsInterface(module, interfaceId)) {
```

Comment:
Wdyt if we check the interface regardless of data length?

Reply:
I thought of this because the usual case for not running `onInstall` is if it's been already initialized, which usually only happens on re-install (i.e. if you're using the same hook id across two different validations), in which case the previous install with a call to `onInstall` would have checked this. Probably should either include that rationale in a comment, or update this to always check. Wdyt?

Reply:
I'm not that partial to one way or another, but I think the assumption we'd always call `onInstall` the first time we install may not necessarily be a safe one. I could imagine modules which inherently don't need to be directly initialized probs.

For that reason always checking makes a bit more sense to me, but it could end up being redundant. I leave this one up to your call!

Reply:
One neat side effect of this, is that this actually lets us still do the "install EOA for direct call validation" - just added a test for this here: https://github.com/erc6900/reference-implementation/pull/147/commits/b858ccb3eb40d932fe1927a33efe89a8b3b1dff2

Given that one use case, I'll keep the interface check behind the initialization data for now.